### PR TITLE
Reduce usage of `SAME` threadpool name

### PR DESCRIPTION
**Pull request by DaveCTurner**
These days in most places where we mention `SAME` we're immediately
looking up the corresponding executor, and therefore can just mention
`EsExecutors#DIRECT_EXECUTOR_SERVICE` directly instead.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java [hunk 1]
```diff
@@ -1196,6 +1198,21 @@ public static String randomDateFormatterPattern() {
         return randomFrom(FormatNames.values()).getName();
     }
 
+    /**
+     * Randomly choose between {@link EsExecutors#DIRECT_EXECUTOR_SERVICE} (which does not fork), {@link ThreadPool#generic}, and one of the
+     * other named threadpool executors.
+     */
+    public static Executor randomExecutor(ThreadPool threadPool, String... otherExecutorNames) {
```
Comment (pxsalehi) [inline test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:1205]: nit/question: is there a use of this that passes anything to `otherExecutorNames`? Having something like `genericOrDirect` for the cases that needs one of the two might make this easier to read.

**Review by pxsalehi — APPROVED** [review]
Is there a special reason to have SAME (probably still used), but are all replaceable like this?

File: test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java (hunk 1 above)
Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:1205]: In this PR there's just one, in `JoinValidationServiceTests`. I expect to be able to adjust `org.elasticsearch.index.shard.IndexShardTests#testRunUnderPrimaryPermitDelaysToExecutorWhenBlocked` to use this too although it's a little more involved so I'll do it later.

**Comment by DaveCTurner** [comment]
> Is there a special reason to have SAME (probably still used), but are all replaceable like this?

Not a _good_ reason to keep `SAME` indeed, although some of the usages are a little trickier to rework than the ones done here. I will follow up at some point.

**Comment by DaveCTurner** [comment]
`SAME` is now wholly unnecessary, https://github.com/elastic/elasticsearch/pull/107249 completes its removal.

**Pull request merged**